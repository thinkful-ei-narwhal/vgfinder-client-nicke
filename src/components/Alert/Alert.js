import React, { Component } from 'react';
import './Alert.css'

export function ShowAlert(props) {
  return (
    <div className={props.show}>
      {props.message}
    </div>)
}

export default class Alert extends Component {
  state = {
    show: "snackbar show"
  }

  componentDidMount() {
    this.timeout = setTimeout(() => {
      this.setState({ show: "snackbar" });
      this.props.handleErrorClick();
    }, 2800)
  }

  render() {
    return (
      <ShowAlert message={this.props.message} show={this.state.show} />
    );
  }
}