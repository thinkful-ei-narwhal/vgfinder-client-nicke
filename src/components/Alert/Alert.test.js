import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Alert from "./Alert";

describe(`Alert component`, () => {
  it("renders an alert by default", () => {
    const wrapper = shallow(<Alert />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders an alert with the correct message of the prop passed in", () => {
    const wrapper = shallow(<Alert message={{ error: "Test Error" }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders an alert with the correct message of the prop passed in, then executes the callback", () => {
    let stateDummy = 1;
    const wrapper = shallow(
      <Alert
        message={{ error: "Test Error" }}
        handleErrorClick={() => {
          stateDummy++;
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    //timeout has to be just a little longer than the timeout inside of the alert
    setTimeout(() => {
      expect(stateDummy).toBe(2);
    }, 3000);
  });
});
