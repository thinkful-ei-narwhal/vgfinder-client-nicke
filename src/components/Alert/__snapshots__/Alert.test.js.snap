// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders an alert by default 1`] = `
<ShowAlert
  show="snackbar show"
/>
`;

exports[`Alert component renders an alert with the correct message of the prop passed in 1`] = `
<ShowAlert
  message={
    Object {
      "error": "Test Error",
    }
  }
  show="snackbar show"
/>
`;

exports[`Alert component renders an alert with the correct message of the prop passed in, then executes the callback 1`] = `
<ShowAlert
  message={
    Object {
      "error": "Test Error",
    }
  }
  show="snackbar show"
/>
`;
